src/ioctls.rs

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module implements Intel SGX-related IOCTLs using the iocuddle crate.
+//! All references to Section or Tables are from
+//! [Intel® 64 and IA-32 Architectures Software Developer’s Manual Volume 3D: System Programming Guide, Part 4](https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-vol-3d-part-4-manual.pdf)
+
+#![allow(dead_code)]
+
+use crate::page::{SecInfo, Secs};
+use crate::signature::Signature;
+
+use core::marker::PhantomData;
+
+pub const ENCLAVE_CREATE: u64 = 0x7ffd84d964e8;
+pub const ENCLAVE_ADD_PAGES: u64 = 0x7ffd84d96450;
+pub const ENCLAVE_INIT: u64 = 0x7ffd84d96528;
+pub const ENCLAVE_RESTRICT_PERMISSIONS: u64 = 0x7ffd84d96910;
```

Comment:
Why aren't you using `iocuddle`?

Reply:
It causes compilation issues with the shim. We are now doing enarxcall but I think it would be wise to keep this dependency free to be future-proof. I think the right question would be to rather ask what would using iocuddle improve? It's as magic number as these, just a syntax tree version of it.

Reply:
I pushed a new version with two other issues fixed. Still waiting for input on this one.

Reply:
@jarkkojs That's why you put it behind a feature that the shim doesn't turn on.

Reply:
fixed